src/container.jl

```diff
@@ -269,10 +265,10 @@ function resample_propagate!(
             pi = particles[i]
             isref = pi === ref
             p = isref ? fork(pi, isref) : pi
-            nseeds = isref ? ni - 1 : ni
+            rng = isref ? particles[j + 1].rng.rng : p.rng.rng
```

Comment:
There might be a subtle issue here related to `j+1`. If the particle `j+1` has non-zero children, then its `rng` would be used by itself, and maybe a child from the `ref` particle.  That would lead to two particles (one ref's child, one particle `j+1`'s child) sharing the same `rng`. Is that an intended behaviour?

Reply:
That was indeed an issue, instead we can pick a RNG a priori and use it to generate keys for the ref particle children. (just to seed the rng, these childrent would still inherit from the reference history)